tests/integration/accountRouter.test.js

```diff
@@ -22,6 +21,17 @@ describe('Account Routes', function () {
         expect(res.statusCode).toBe(200)
     })
 
+    test('responds with OK to provided parameters', async () => {
+        const response = await testSession.get('/account/password/confirmReset?username=turbo2&token=XXXXX')
+        expect(response.statusCode).toBe(200)
+    })
+
+    test('redirect to reset request page if missing parameters with flash parameter', async () => {
+        const response = await testSession.get('/account/password/confirmReset')
+        expect(response.statusCode).toBe(302)
+        expect(response.headers.location).toContain('/account/requestPasswordReset?flash=')
```

Comment:
if you follow the request, you could expect for the "invalid request" and get the other code covered.

Reply:
Does not appear that following the request is configured as a standard option nor done as a current standard.

I would think i could test it using `response.res.locals` but that's undefined

Reading from the text response you could check for `<ul class=\"flash-errors\">` but thats present regardless of the case and is empty.

Potentially readable from session cooking but hit a wall there. 

Will revisit this tomorrow.

Reply:
the website is broken everywhere, created a solution for your problem:
https://github.com/beckpaul/faf-website/pull/1/files

in general: flash messages are broken, they often have not the right structure (someone changed the layout)
tests: i restricted the tests and disallowed non localhost requests, so every request done needs to have a mock if not to localhost